app/services/mistral_provider.py

```
@@ -0,0 +1,130 @@
+                    raise ProviderError(self.name, f"API error: {response.status_code}")
+                
+                data = response.json()
+                translation = data["choices"][0]["message"]["content"].strip()
+                
+                if not translation:
+                    raise ProviderError(self.name, "Empty translation response")
+                    
+                return translation
+                
+        except httpx.HTTPError as e:
+            raise ProviderError(self.name, f"HTTP error: {str(e)}", e)
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)